### PR TITLE
Implement swipe navigation

### PR DESCRIPTION
**Pull request by NguyenDytrich**
Swipe navigation that follows the same direction indicated by the navbar buttons.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,5 +13,4 @@ xcuserdata/
 Podfile.lock
 Pods/*
 build/
-*.DS_Store
```
Comment (mamaral) [inline .gitignore]: What was the reasoning for removing this?

**Comment by mamaral** [comment]
This looks sweet!

File: .gitignore (hunk 1 above)
Comment (NguyenDytrich) [inline .gitignore]: It was a duplicate entry; it's mentioned earlier in the file.

Comment (mamaral) [inline .gitignore]: Ah, it was indeed! Thanks for that!

**Comment by The-Penultimate-Defenestrator** [comment]
What's the spade of time between a PR being merged and the update appearing on the App Store?

**Comment by mamaral** [comment]
@The-Penultimate-Defenestrator I'll try and get a new version pushed to the app store sometime this week - so it will likely be a few weeks before its available on the store.

**Comment by The-Penultimate-Defenestrator** [comment]
Which will also hopefully fix the recent issue with comics … ?

**Comment by mamaral** [comment]
The comics issue is a db issue and should be resolved whenever @fergusean gets around to it. I'll try and get that done ASAP.

**Comment by The-Penultimate-Defenestrator** [comment]
Does the app not just scrape the website? That would make sense to me… You could also just have the app skip comics that throw an error if that were the case.

**Comment by mamaral** [comment]
No, we have a server that grabs content from the json API itself. Having our own server allows us to do a few things, like pre-calculate image aspect ratios for the mosaic tile design, send push notifications, etc.

**Comment by The-Penultimate-Defenestrator** [comment]
Ah. Neat. Still, maybe there should be a scraping mechanism as a backup, so
that not *all* functionality is lost if the server goes down.

Do you mean that there is an XKCD JSON API, or is that something you guys
implemented?

**Comment by mamaral** [comment]
@The-Penultimate-Defenestrator Here's the info for their "API" https://xkcd.com/json.html

Every now and then the comics are something other than images, they might be little interactive browser mini-games or something, so those tend to cause the server to hiccup and we generally just throw a static image as a placeholder in there for that comic. 

We still have to determine what we want to do for those and do it, but it always comes down to this being a free, ad-free, open-source, side-project (for fun), so this tends to be lower priority than other projects/jobs I'm working on. We should have a fix for the server shortly, and I'll keep you updated on the next release.

**Comment by The-Penultimate-Defenestrator** [comment]
Great. Is the server's software open source as well? I could imagine some
simple exception handling that would automatically capture a screenshot for
interactive comics…

**Comment by mamaral** [comment]
https://github.com/mamaral/xkcd-Open-Source/tree/master/cache-server

**Comment by The-Penultimate-Defenestrator** [comment]
Cool. I know a bit of PHP, maybe I'll submit a PR if I have some extra
time.

**Comment by NguyenDytrich** [comment]
The `img` attribute in the JSON for comic 1663 doesn't link to a supported image file (jpeg, png, gif). This raises an uncaught exception causing the cache server to crash.

**Comment by The-Penultimate-Defenestrator** [comment]
That should be easy to catch, the server might even be able to generate an
image with placeholder text.

**Comment by mamaral** [comment]
@NguyenDytrich Yes, we've determined this was the case already, and we probably should have updated the open issue https://github.com/mamaral/xkcd-Open-Source/issues/39 with those findings. Let's move discussion over there.

**Pull request merged**